docs/en/getting-started/example-datasets/laion.md

```diff
@@ -0,0 +1,259 @@
+│ https://www.baxterboo.com/images/breeds/medium/cairn-terrier.jpg                                                                                                                     │ Cairn Terrier Photo                                                  │
+└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────────────────────────┘
+
+8 rows in set. Elapsed: 0.641 sec. Processed 22.06 thousand rows, 49.36 MB (91.53 thousand rows/s., 204.81 MB/s.)
+```
+
+The speed has increased significantly. But now the results sometimes differ from what you are looking for. This is due to the approximation of the search and the quality of the constructed embedding. Note that the example was given for picture embeddings, but there are also text embeddings in the dataset, which can also be used for searching.
+
+## Scripts for embeddings
+
+Usually we do not want to get embeddings from existing data, but to get them for new data and look for similar ones in old data. We can use [UDF](../../sql-reference/functions/index.md#sql-user-defined-functions) for this purpose. They will allow you to set the `target` vector without leaving the client. All of the following scripts will be written for the `ViT-B/32` model as it was used for this dataset. You can use any model, but it is necessary to build embeddings in the dataset and for new objects using the same model.
+
+### Text embeddings
+
+`encode_text.py`:
+```python
+#!/usr/bin/python3
+import clip
+import torch
+import numpy as np
+import sys
+
+if __name__ == '__main__':
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    model, preprocess = clip.load("ViT-B/32", device=device)
+    for text in sys.stdin:
+        inputs = clip.tokenize(text)
+        with torch.no_grad():
+            text_features = model.encode_text(inputs)[0].tolist()
+        sys.stdout.flush()
+```
+
+`encode_text_function.xml`:
+```xml
+<functions>
+    <function>
+        <type>executable</type>
+        <name>encode_text</name>
+        <return_type>Array(Float32)</return_type>
+        <argument>
+            <type>String</type>
+            <name>text</name>
+        </argument>
+        <format>TabSeparated</format>
+        <command>encode_text.py</command>
+        <command_read_timeout>1000000</command_read_timeout>
+    </function>
+</functions>
+```
+
+Now we can simply use:
+
+```sql
+SELECT encode_text('cat');
+```
+
+The first use will be slow because you need to load the model. But repeated queries will be fast. Then we copy the results to ``set param_target=...`` and can easily write queries 
+
+### Image embeddings
+
+For pictures it is the same, but you do not need to send the picture but the path (if necessary, you can implement a download picture with processing, but it will be longer)
```

Comment:
```suggestion
For pictures, the process is similar, but you send the path instead of the picture (if necessary, you can implement a download picture with processing, but it will take longer)
```